platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm

```diff
@@ -318,14 +318,46 @@ - (id)mgl_jsonExpressionObject {
         case NSNotEqualToPredicateOperatorType:
             op = @"!=";
             break;
+        case NSBetweenPredicateOperatorType: {
+            op = @"all";
+            NSArray *limits = self.rightExpression.constantValue;
+            NSPredicate *leftHandPredicate = [NSComparisonPredicate predicateWithLeftExpression:limits[0]
+                                                                                rightExpression:self.leftExpression
+                                                                                       modifier:NSDirectPredicateModifier
+                                                                                           type:NSLessThanOrEqualToPredicateOperatorType
+                                                                                        options:0];
+            NSPredicate *rightHandPredicate = [NSComparisonPredicate predicateWithLeftExpression:self.leftExpression
+                                                                                 rightExpression:limits[1]
+                                                                                        modifier:NSDirectPredicateModifier
+                                                                                            type:NSLessThanOrEqualToPredicateOperatorType
+                                                                                         options:0];
+            return @[op, leftHandPredicate.mgl_jsonExpressionObject, rightHandPredicate.mgl_jsonExpressionObject];
+        }
+        case NSInPredicateOperatorType: {
+            NSMutableArray *elements = [NSMutableArray arrayWithObjects:@"match", self.leftExpression.mgl_jsonExpressionObject, nil];
+            NSArray *optionsExpressions = self.rightExpression.constantValue;
+            NSEnumerator *optionsEnumerator = optionsExpressions.objectEnumerator;
+            while (id object = optionsEnumerator.nextObject) {
```

Comment:
A `while` loop with an NSEnumerator is only useful when you need to also work with indices, which isn’t the case here. (And in that case, you could use the more modern `-enumerateObjectsAtIndexes:options:usingBlock:` anyways.) Replace the `while` loop with a `for`-`in` loop, which takes advantage of fast enumeration.